projects/github-actions/repo-gardening/src/tasks/check-description/index.js

```diff
@@ -446,6 +479,64 @@ async function updateLabels( payload, octokit ) {
 	}
 }
 
+/**
+ * Based off the changelog entries committed with the PR,
+ * let's attempt to infer the type of the PR.
+ * If we can infer a type, we'll add a label to the PR, matching that type.
+ * If we cannot, we'll leave the PR as is.
+ *
+ * @param {WebhookPayloadPullRequest} payload          - Pull request event payload.
+ * @param {GitHub}                    octokit          - Initialized Octokit REST client.
+ * @param {Array}                     changelogEntries - List of changelog entries.
+ *
+ * @return {Promise<boolean>} Promise resolving to boolean ; true if we were able to infer a type, false otherwise.
+ */
+async function inferType( payload, octokit, changelogEntries ) {
+	const {
+		repository: {
+			owner: { login: ownerLogin },
+			name: repo,
+		},
+		pull_request: { number },
+	} = payload;
+
+	// Extract the type of the PR from the changelog entries.
+	let type = '';
+	for ( const entry of changelogEntries ) {
+		if ( entry.changeType ) {
+			type = entry.changeType;
```

Comment:
What if different entries have different types? For example, 'added' something to one package, then in another package or plugin a 'fixed' making use of the new function to fix a bug?

As written, it'd pick one of those at random to use as the type.

Reply:
Yes, that was my biggest question about this change. I don't have a perfect solution for that I'm afraid.